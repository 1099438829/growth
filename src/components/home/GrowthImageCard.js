/* eslint-disable global-require,import/no-dynamic-require */
import React from 'react';
import { Image, Text, TouchableHighlight, View } from 'react-native';
import PropTypes from 'prop-types';

const GrowthImageCard = props => (
  <TouchableHighlight
    activeOpacity={1}
    onPress={() => props.action()}
  >
    <Image
      source={props.imageUrl}
      resizeMode="cover"
      borderRadius={6}
      style={{ marginLeft: 20, marginRight: 20, marginTop: 10, width: null, height: 120 }}
    >
      <View style={{ height: 40, opacity: 0.8, top: 80, backgroundColor: '#fff' }}>
        <Text style={{ textAlign: 'center', fontSize: 20, top: 8, color: '#000' }}> { props.text }</Text>
      </View>
    </Image>
  </TouchableHighlight>
);

GrowthImageCard.propTypes = {
  imageUrl: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  action: PropTypes.func.isRequired,
};

export default GrowthImageCard;
