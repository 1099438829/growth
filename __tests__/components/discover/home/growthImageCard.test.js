/* eslint-disable global-require */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import GrowthImageCard from '../../../../src/components/home/GrowthImageCard';


it('renders correctly', () => {
  // eslint-disable-next-line no-unused-vars
  const dialog = renderer.create(
    <GrowthImageCard
      imageUrl={require('../../../../assets/growth-ui/img/home-6.jpg')}
      actions={() => {}}
      text={'Growth 技能树'}
    />,
  );
  const dialogJson = dialog.toJSON();
  expect(dialogJson).toMatchSnapshot();
});
