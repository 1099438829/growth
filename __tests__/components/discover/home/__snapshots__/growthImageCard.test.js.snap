// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    borderRadius={6}
    resizeMode="cover"
    source={1}
    style={
      Object {
        "height": 120,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 10,
        "width": null,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "height": 40,
          "opacity": 0.8,
          "top": 80,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 20,
            "textAlign": "center",
            "top": 8,
          }
        }
      >
         
        Growth 技能树
      </Text>
    </View>
  </Image>
</View>
`;
